cpp/harvest/nacjd.cc

```diff
@@ -355,28 +385,41 @@ void ParseJSONAndWriteMARC(MARC::Writer * const title_writer) {
 
 
 int Main(int argc, char *argv[]) {
-    if (argc < 3 or argc > 4)
+    if (argc < 3)
         Usage();
 
     unsigned count_rows(9000);
-    if (argc == 4) {
-        count_rows = std::stoi(argv[3]);
+    unsigned mode_param(0);
+    std::string mode("");
+
+    if (argv[3]) {
+        mode = argv[3];
+    }
+
+    if (mode == "count_of_rows") {
+        count_rows = std::stoi(argv[4]);
```

Comment:
This whole block should be moved to an "else if" block where all modes are checked (`if (mode == "get_by_ID") {`... `} else if mode == "count_of_rows" {`). And also the declaration of tje count_rows variable should be moved into this block because we do not need the variable outside of the block.